src/web/components/AuditTrail/AuditTrailTable.tsx

```diff
@@ -0,0 +1,124 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { useState } from 'react';
+
+import { AuditTrailDTO } from '../../../api/entities/AuditTrail';
+import { SortableProvider, useSortable } from '../../contexts/SortableTableProvider';
+import { PagingTool } from '../Core/Paging/PagingTool';
+import { RowsPerPageValues } from '../Core/Paging/PagingToolHelper';
+import { SortableTableHeader } from '../Core/Tables/SortableTableHeader';
+import { TableNoDataPlaceholder } from '../Core/Tables/TableNoDataPlaceholder';
+import AuditLog from './AuditLog';
+
+import './AuditTrail.scss';
+
+type AuditLogTableProps = Readonly<{
+  auditTrail: AuditTrailDTO[];
+}>;
+
+function AuditLogTable({ auditTrail }: AuditLogTableProps) {
+  const initialRowsPerPage = 25;
+
+  const [rowsPerPage, setRowsPerPage] = useState<RowsPerPageValues>(initialRowsPerPage);
+  const [pageNumber, setPageNumber] = useState<number>(1);
+  const [searchText, setSearchText] = useState('');
+
+  const getPagedAuditRows = (values: AuditTrailDTO[]) => {
+    const pagedRows = values.filter((_, index) => {
+      return (
+        index >= (pageNumber - 1) * rowsPerPage &&
+        index < (pageNumber - 1) * rowsPerPage + rowsPerPage
+      );
+    });
+    return pagedRows;
+  };
+
+  const onChangeDisplayedAuditRows = (
+    currentPageNumber: number,
+    currentRowsPerPage: RowsPerPageValues
+  ) => {
+    setPageNumber(currentPageNumber);
+    setRowsPerPage(currentRowsPerPage);
+  };
+
+  const handleSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchText(event.target.value);
+    setPageNumber(1);
+    setRowsPerPage(initialRowsPerPage);
+  };
+
+  // first search
+  let searchedAuditRows = auditTrail;
+  if (searchText.length > 2) {
+    searchedAuditRows = auditTrail.filter((item) => {
+      const search = searchText.toLowerCase();
+      return (
+        item.userEmail.toLowerCase().indexOf(search) >= 0 ||
+        item.event.toLowerCase().indexOf(search) >= 0 ||
+        JSON.stringify(item.eventData).toLowerCase().indexOf(search) >= 0
+      );
+    });
+  }
+  // then sort
+  const { sortData } = useSortable<AuditTrailDTO>();
+  const sortedAuditLogs = sortData(searchedAuditRows);
+  // finally page
+  const pagedRows = getPagedAuditRows(sortedAuditLogs);
+
+  return (
+    <div className='audit-trail-table-container'>
+      <div className='audit-trail-table-header'>
+        <div className='audit-trail-table-header-right'>
+          <div className='audit-trail-search-bar-container'>
+            <input
+              type='text'
+              className='audit-trail-search-bar'
+              onChange={handleSearch}
+              placeholder='Search Audit Trail'
```

Comment:
suggestion: use 'Search audit trail' to align with other searches (e.g. cstg tables)